src/lizaalert/courses/permissions.py

```diff
@@ -22,3 +25,31 @@ def has_object_permission(self, request, view, obj):
             if current_lesson.ordering >= obj.ordering:
                 return True
         return False
+
+
+class EnrolledAndCourseHasStarted(permissions.BasePermission):
+    """
+    Доступ к урокам курса только для записанных и на курс, который начался.
+
+    access_statuses - статусы пользователся, с которыми можно получить доступ к курсу.
+    update_subscriptions - обновляет статусы подписок пользователя.
+    """
+
+    message = "Доступ к урокам курса только для подписчиков и на курс, который начался."
+
+    def has_object_permission(self, request, view, obj):
+        user = request.user
+        course = obj.chapter.course
+        update_one_subscription(user, course)
+        access_statuses = [
+            Subscription.Status.AVAILABLE,
+            Subscription.Status.IN_PROGRESS,
+            Subscription.Status.COMPLETED,
+        ]
+        if user.is_authenticated:
+            try:
+                status = user.subscriptions.get(course=course)
+                return status.status in access_statuses
+            except Exception:
+                return False
```

Comment:
Здесь по идее можно немного оптимизировать запрос
```suggestion
if user.subscriptions.filter(course=course, status__in=access_statuses).exists():
        return True
```

Reply:
Топчик, можно даже еще лаконичнее:
```suggestion
        if user.is_authenticated:
            try:
                return user.subscriptions.filter(course=course, status__in=access_statuses).exists()
```